### ISSUE TITLE
Batch consuming + Push

### ISSUE THREAD

**Issue by geekio** [adjust/rmq#45]
Hi,

I'm probably missing something again, please advise. Why not expose a wrapper for Delivery.Push() when it comes to batch consuming through rmq.Deliveries ? It only got Ack and Reject.

Thanks!

p.s. Just to mention, using PANIC makes the code a bit tricky to use. I am using the lib as an add-on, it should not interrupt the runtime. For now I wrapped some calls (e.g. openConnection) in defer+recover stuff, still does not look fine.

**Comment by wellle** [comment]
> Why not expose a wrapper for Delivery.Push() when it comes to batch consuming through rmq.Deliveries ?

Good idea! Do you want to open a pull request to add `Deliveries.Push()`, similar to Ack and Reject?

---
> using PANIC makes the code a bit tricky to use.

Yes, I'm aware. The problem is how error handling should happen exactly. Some of the places in which we panic are fairly disconnected from the direct interface. And we would need to change the interface in many places to expose errors instead of panicking. So this would be a breaking change which I avoided so far. But since we started tagging versions recently (and vendoring is more popular nowadays) I guess we could consider that again. See #8
